ngraph/core/src/op/selu.cpp

```diff
@@ -5,65 +5,58 @@
 #include "ngraph/op/selu.hpp"
 #include "itt.hpp"
 
-#include "ngraph/op/add.hpp"
-#include "ngraph/op/constant.hpp"
-#include "ngraph/op/exp.hpp"
-#include "ngraph/op/maximum.hpp"
-#include "ngraph/op/minimum.hpp"
-#include "ngraph/op/multiply.hpp"
-#include "ngraph/op/subtract.hpp"
-
 using namespace std;
 using namespace ngraph;
 
-NGRAPH_SUPPRESS_DEPRECATED_START
-
-constexpr NodeTypeInfo op::v0::Selu::type_info;
-
-op::v0::Selu::Selu()
-    : FusedOp()
-{
-}
+NGRAPH_RTTI_DEFINITION(op::v0::Selu, "Selu", 0);
 
 op::v0::Selu::Selu(const Output<Node>& data, const Output<Node>& alpha, const Output<Node>& lambda)
-    : FusedOp({data, alpha, lambda})
+    : Op({data, alpha, lambda})
 {
     constructor_validate_and_infer_types();
 }
 
-void ngraph::op::v0::Selu::pre_validate_and_infer_types()
+void op::v0::Selu::validate_and_infer_types()
+
 {
-    set_output_type(0, get_input_element_type(0), get_input_partial_shape(0));
+    NGRAPH_OP_SCOPE(v0_Selu_validate_and_infer_types);
+    auto data_et = get_input_element_type(0);
+    auto alpha_et = get_input_element_type(1);
+    auto lambda_et = get_input_element_type(2);
+
+    // set_output_type(0, get_input_element_type(0), get_input_partial_shape(0));
+    NODE_VALIDATION_CHECK(this,
+                          data_et.is_real() && alpha_et.is_real() && lambda_et.is_real(),
+                          "The data type for input, alpha and lambda is expected to be a floating "
+                          "point type. Got data: ",
+                          data_et,
+                          ", alpha: ",
+                          alpha_et,
+                          ", lambda: ",
+                          lambda_et);
+
+    NODE_VALIDATION_CHECK(
+        this,
+        data_et == alpha_et && alpha_et == lambda_et,
```

Comment:
This is not a good practice, if some external plugin would have dynamic element type, this check could throw exception. To avoid this we check if we can merge types instead, here an [example](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/src/validation_util.cpp#L600)